src/fromager/sources.py

```diff
@@ -326,17 +336,30 @@ def _apply_patch(patch: pathlib.Path, source_root_dir: pathlib.Path):
 
 
 def _warn_for_old_patch(
-    source_root_dir: pathlib.Path, patch_filename: pathlib.Path
+    source_root_dir: pathlib.Path,
+    patches_dir: pathlib.Path,
 ) -> None:
-    # Get the existing version and req name as per the source_root_dir naming conventions
+    # Get the req name as per the source_root_dir naming conventions
     req_name = source_root_dir.name.rsplit("-", 1)[0]
-    existing_version = Version(source_root_dir.name.rsplit("-", 1)[1])
 
-    # Check to see if patches for older version exists
-    if patch_filename.parent.name == source_root_dir.name:
-        logger.warning(
-            f"{req_name}: patches for version {existing_version!s} exist but will not be applied"
+    # Get all the patches for that requirement
+    req_patches = overrides.patches_for_req(patches_dir, source_root_dir.name)
+
+    if len(req_patches) > 0:
+        # Set up regex to filter out unwanted patches.
+        # Example: instructlab and instructlab-sdg
+        pattern = re.compile(rf"^{req_name}(-v?\d+\.\d+\.\d+)?$")
+
+        filtered_patches = [s for s in req_patches if pattern.match(s.name)]
+
+        current_path = source_root_dir.parent.joinpath(
+            "patches_dir", source_root_dir.name
         )
+        # If filtered patches contain patches for current version, remove the path because we dont want to warn
+        if current_path in filtered_patches:
+            filtered_patches.remove(current_path)
+        if len(filtered_patches) > 0:
+            logger.warning(f"{req_name}: old patches exist but will not be applied")
```

Comment:
They might be patches for a newer version, so the message shouldn't assume which one is the newer/older. It would be good to log each patch name that is being ignored.

Reply:
Okay, I am now logging` filtered_patches`. 
Should I use `iterdir()` to log individual patch files?